### PR TITLE
add python codecov

### PR DESCRIPTION
**Pull request by guihao-liang**
add pytest-cov plugin to generate the report for the test coverage for our python source file. This test coverage doesn't cover C++ implementation code.

Sample output.
```
src/python/turicreate/_connect/main.py                                                    29     29     0%
src/python/turicreate/_gl_pickle.py                                                      165    165     0%
src/python/turicreate/_json.py                                                            16     16     0%
src/python/turicreate/_scripts/_pylambda_worker.py                                        56     56     0%
src/python/turicreate/_sys_util.py                                                       221    221     0%
src/python/turicreate/aggregate.py                                                        49     49     0%
src/python/turicreate/data_structures/gframe.py                                          163    163     0%
src/python/turicreate/data_structures/grouped_sframe.py                                   34     34     0%
src/python/turicreate/data_structures/image.py                                            81     81     0%
src/python/turicreate/data_structures/sarray.py                                          820    820     0%
src/python/turicreate/data_structures/sarray_builder.py                                   34     34     0%
src/python/turicreate/data_structures/sframe.py                                         1450   1450     0%
src/python/turicreate/data_structures/sframe_builder.py                                   56     56     0%
src/python/turicreate/data_structures/sgraph.py                                          369    369     0%
src/python/turicreate/data_structures/sketch.py                                           98     98     0%
...
```

### CODE REVIEW

**Comment by guihao-liang** [comment]
downstream of #2973

File: scripts/run_python_test.sh [hunk 1]
```diff
@@ -18,9 +18,9 @@ WORKSPACE=${SCRIPT_DIR}/..
 
 unset PYTHONPATH
 
-cd $WORKSPACE
+cd "$WORKSPACE"
```
Comment (TobyRoseman) [inline scripts/run_python_test.sh]: Is `"$WORKSPACE"` the same as `${WORKSPACE}`?

Comment (guihao-liang) [reply scripts/run_python_test.sh]: No, the quotes are intentional. https://github.com/koalaman/shellcheck/wiki/SC2086

Comment (guihao-liang) [reply scripts/run_python_test.sh]: I suggest you use this tool called `shellcheck` along with `pre-commit`.

File: scripts/run_python_test.sh [hunk 2]
```diff
@@ -44,15 +44,16 @@ if ! type "parallel" 2> /dev/null; then
         if [[ $OSTYPE != msys ]]; then
           cmd=${PYTHON_EXECUTABLE}
         fi
-        cmd="${cmd} ${PYTEST_EXECUTABLE} -v $PYTHONPATH/turicreate/test --junit-xml=alltests.pytest.xml"
-        echo $cmd
+        cmd="${cmd} ${PYTEST_EXECUTABLE} --cov -v $PYTHONPATH/turicreate/test \
+          --junit-xml=alltests.pytest.xml"
+        echo "$cmd"
```
Comment (TobyRoseman) [inline scripts/run_python_test.sh]: You probably want to remove this `echo`. Especially since `set -x` is included at the top of the script.

Comment (guihao-liang) [reply scripts/run_python_test.sh]: That's right! will do.

**Comment by guihao-liang** [comment]
pass internal tests.

**Pull request merged**